src/Components/CalculatorButtons.js

```diff
@@ -1,27 +1,36 @@
+// CalculatorButtons.js
+
 import React from 'react';
 
-function CalculatorButtons() {
+function CalculatorButtons({
+  onNumberClick,
+  onOperatorClick,
+  onEqualClick,
+  onClearClick,
+  onDecimalClick,
+  onToggleSign,
+}) {
   return (
     <div className="calculator-grid">
-      <button type = 'button'>AC</button>
-      <button type = 'button'>+/-</button>
-      <button type = 'button'>%</button>
-      <button type='button' className="oper">/</button>
-      <button type = 'button'>7</button>
-      <button type = 'button'>8</button>
-      <button type = 'button'>9</button>
-      <button className="oper">*</button>
-      <button type = 'button'>4</button>
-      <button type = 'button'>5</button>
-      <button type = 'button'>6</button>
-      <button className="oper">-</button>
-      <button type = 'button'>1</button>
-      <button type = 'button'>2</button>
-      <button type = 'button'>3</button>
-      <button className="oper">+</button>
-      <button className="span-two">0</button>
-      <button type = 'button'>.</button>
-      <button type='button' className="oper">=</button>
+      <button type="button" onClick={onClearClick}>AC</button>
+      <button type="button" onClick={onToggleSign}>+/-</button>
+      <button type="button" onClick={() => onOperatorClick('%')}>%</button>
+      <button type="button" className="oper" onClick={() => onOperatorClick('/')}>/</button>
+      <button type="button" onClick={() => onNumberClick(7)}>7</button>
+      <button type="button" onClick={() => onNumberClick(8)}>8</button>
+      <button type="button" onClick={() => onNumberClick(9)}>9</button>
+      <button className="oper" onClick={() => onOperatorClick('*')}>*</button>
+      <button type="button" onClick={() => onNumberClick(4)}>4</button>
+      <button type="button" onClick={() => onNumberClick(5)}>5</button>
+      <button type="button" onClick={() => onNumberClick(6)}>6</button>
+      <button className="oper" onClick={() => onOperatorClick('-')}>-</button>
+      <button type="button" onClick={() => onNumberClick(1)}>1</button>
+      <button type="button" onClick={() => onNumberClick(2)}>2</button>
+      <button type="button" onClick={() => onNumberClick(3)}>3</button>
+      <button className="oper" onClick={() => onOperatorClick('+')}>+</button>
+      <button className="span-two" onClick={() => onNumberClick(0)}>0</button>
+      <button type="button" onClick={onDecimalClick}>.</button>
+      <button type="button" className="oper" onClick={onEqualClick}>=</button>
```

Comment:
- You have done well so far, but there are some issues to be fixed. Each time I click on an operator, it shows as a `0` on the screen, this can be confusing for users. I suggest you fix this issue 🙏 


https://github.com/Zel-hub7/math-magicians/assets/73607512/1b9c7c99-abf5-4194-a123-7cbbcce1445c